lib/Bagger/Storage/LenkwerkSetup.pm

```diff
@@ -173,6 +181,89 @@ sub dbi_str { "DBI:Pg:" .
                        _dbi_str_keyval($dbport, 'port'),
 	 )
 }
+
+=item dsn_uri
+
+   Returns a well-formed URI from config input.  All items are properly
+   escaped.
+
+=cut
+
+sub dsn_uri {
+    my $dsn = 'postgresql://';
+    $dsn .= join ':', (uri_escape_utf8($dbuser // ''),
+                       uri_escape_utf8($dbpass // ''));
+    $dsn .= '@';
+    $dsn .= join ':', (uri_escape_utf8($dbhost // ''),
+                       uri_escape_utf8($dbport // ''));
+    $dsn .= '/';
+    $dsn .= uri_escape_utf8($lenkwerkdb);
+    return $dsn;
+}
+
+
+# use for programs that don't support dns's
+sub _cli_args {
+    return (
+        ($dbuser ? ('-U', "$dbuser") : ()),
+        ($dbhost ? ('-h', "$dbhost") : ()),
+        ($dbport ? ('-p', $dbport) : ()),
+    );
+}
+
+=head1 DATABASE SETUP ROUTINES
+
+=head2 createdb
+
+  This function creates a database and returns 1 if successful and 0 if not.
+
+  It thrown an exception if the database name has not been specified.
+
+=cut
+
+sub createdb {
+    croak 'No database set' unless $lenkwerkdb;
+    local $! ; # mask last system error
+
+    # If we are told to use a password, then we will
+    # mask the environment variable in this routine.
+    #
+    # This is safer than alternatives.
+
+    local $ENV{PGPASSWORD} = $dbpass if defined $dbpass;
+    my $failure = system('createdb', _cli_args, $lenkwerkdb);
+    warn 'Database creation failed' if $failure;
```

Comment:
As a potential admin, I'd like to know how exactly this fails without attaching strace.
Not an urgent thing to do, but I'd like each fork to log errno, exitval and potentially the signal. When debugging, such info is gold.
We don't even have logging yet, so: not urgent.

Reply:
Currently I think stderr and stdout get sent to relevant Perl channels if I understnad correctly.

However, I agree we need to abstract this.  I have added ticket #30  to address